modules/oadp-monitoring.adoc

```diff
@@ -0,0 +1,383 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/troubleshooting.adoc
+
+:_content-type: CONCEPT
+[id="oadp-monitoring_{context}"]
+= OADP Monitoring
+
+The {product-title} provides a xref:../../monitoring/monitoring-overview.adoc#about-openshift-monitoring[monitoring stack] that allows users and administrators to effectively monitor and manage their OpenShift clusters, as well as monitor and analyze the workload performance of user applications and services running on the clusters, including receiving alerts if an event occurs.
+
+The OADP Operator leverages an OpenShift xref:../../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects_enabling-monitoring-for-user-defined-projects[User Workload Monitoring] provided by the OpenShift Monitoring Stack for retrieving metrics from the Velero service endpoint. The monitoring stack allows creating user-defined Alerting Rules or querying metrics, using the OpenShift Metrics query front-end.
+
+With enabled User Workload Monitoring it is possible to configure and use any Prometheus-compatible third-party UI, such as Grafana to visualize Velero metrics.
+
+[id="OADP-monitoring-setup_{context}"]
+== OADP monitoring setup
+Monitoring metrics requires enabling monitoring for the user-defined projects and creating a ServiceMonitor resource to scrape those metrics from the already enabled OADP service endpoint that resides in the `openshift-adp` namespace.
+
+[NOTE]
+====
+The configuration files that enable User Workload Monitoring can be found https://github.com/openshift/oadp-operator/tree/master/docs/examples/manifests/user_monitoring[here], however they will override any previous configuration that could have been on the cluster, so use them with caution. They are named with a number prefix, which represents the order in which they should be applied.
+====
+
+.Prerequisites
+* You created an OADP operator, a credentials secret, and a DataProtectionApplication (DPA) custom resource (CR).
+* Your shell has the `oc` CLI command available.
+* You have Administrator level access to the OpenShift cluster.
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` ConfigMap object in the `openshift-monitoring` namespace and add or enable the `enableUserWorkload` option under `data/config.yaml`.
++
+[source,terminal]
+----
+$ oc edit configmap cluster-monitoring-config -n openshift-monitoring
+----
+[source,yaml]
++
+----
+apiVersion: v1
+data:
+  config.yaml: |
+    enableUserWorkload: true  # Add this option or set to true
+kind: ConfigMap
+metadata:
+# [...]
+----
++
+. Wait a short period of time to verify the User Workload Monitoring Setup by checking if the following components are up and running in the `openshift-user-workload-monitoring` namespace.
```

Comment:
What is a short period of time? Can we write in minutes, for example several minutes?

Reply:
It really depends on the cluster, but my estimate is somewhere between few seconds up to 2-3 minutes. I was trying to find a single command which will watch a particular resource to be ready, however we have here multiple of those so it's not that simple.

See other part of the documentation which is pretty similar and that wording in point number 4:

https://62710--docspreview.netlify.app/openshift-enterprise/latest/monitoring/enabling-monitoring-for-user-defined-projects.html#enabling-monitoring-for-user-defined-projects_enabling-monitoring-for-user-defined-projects